src/harness/reference_models/pre_iap_filtering/zone_purge_test.py

```diff
@@ -0,0 +1,102 @@
+   # Load PPA record
+   gwpz_record = json.load(
+     open(os.path.join('testdata', 'testset1', 'gwpz_0.json')))
+   ppa_record = json.load(
+     open(os.path.join('testdata', 'testset1', 'ppa_0.json')))
+   pal_record_0 = json.load(
+     open(os.path.join('testdata', 'testset1', 'pal_0.json')))
+   pal_record_1 = json.load(
+     open(os.path.join('testdata', 'testset1', 'pal_1.json')))
+   pal_record_2 = json.load(
+     open(os.path.join('testdata', 'testset1', 'pal_2.json')))
+
+   pal_record_list = [pal_record_0, pal_record_1, pal_record_2]
+   fss_record = json.load(
+     open(os.path.join('testdata', 'testset1', 'fss_0.json')))
+   gwbl_record = json.load(
+     open(os.path.join('testdata', 'testset1', 'gwbl_0.json')))
+    
+   ppa_record, pal_records = makePpaAndPalRecordsConsistent(ppa_record, pal_record_list,
+                                                                  'test_user_1')
+   ppa_record['ppaInfo']['cbsdReferenceId'] = ["cbsd_5", "cbsd_3"]
+   fad_object_1 = full_activity_dump.FullActivityDump({'cbsd': [cbsd_0, cbsd_1]})
+   fad_object_2 = full_activity_dump.FullActivityDump({'cbsd': [cbsd_2, cbsd_3]})
+   fad_object_3 = full_activity_dump.FullActivityDump({'cbsd': [cbsd_4, cbsd_5]})
+
+   sas_uut_fad = fad_object_1
+   sas_test_harness_fads = [fad_object_2, fad_object_3]
+
+   protected_entities = {'gwpzRecords': [gwpz_record],'palRecords': pal_records,
+                         'ppaRecords': [ppa_record], 'fssRecords':[fss_record],
+                         'gwblRecords':[gwbl_record]}
+   print "================CBSD Grants passed as input======================"
+   initial_grants = 0
+   for records in sas_uut_fad.getCbsdRecords():
+    for grants in records['grants']:
+      print " ",json.dumps(grants['id'])
+      initial_grants = initial_grants + 1
+   for fad in sas_test_harness_fads:
+     for rec in fad.getCbsdRecords():
+       for grants in rec['grants']:
+         print " ",json.dumps(grants['id'])
+         initial_grants = initial_grants + 1
+   print "===================================================================="
+   zone_purge.zonePurgeReferenceModel(sas_uut_fad, sas_test_harness_fads,\
+             protected_entities['ppaRecords'], protected_entities['palRecords'],\
+             protected_entities['gwpzRecords'], protected_entities['fssRecords'])
+   print "================CBSD Grants received as output======================"
+   final_grants = 0
+   for records in sas_uut_fad.getCbsdRecords():
+     for grants in records['grants']:
+       print " ",json.dumps(grants['id'])
+       final_grants = final_grants + 1
+   for fad in sas_test_harness_fads:
+     for rec in fad.getCbsdRecords():
+       for grants in rec['grants']:
+         print " ",json.dumps(grants['id'])
+         final_grants = final_grants + 1
+   print "===================================================================="
+   self.assertLess(final_grants, initial_grants)
```

Comment:
A little light in terms of unit testing, but I guess we said that it will be improved on another PR.

Reply:
sure,   Will be taken care as part of unit testing PR